SQLQuery_2.sql

```diff
@@ -0,0 +1,295 @@
+INSERT INTO Bills
+VALUES (431, '2023-01-11',1,5);
+INSERT INTO Bills
+VALUES (122, '2022-08-01',0,6);
+INSERT INTO Bills
+VALUES (215, '2022-11-01',1,7);
+INSERT INTO Bills
+VALUES (543, '2023-08-16',0,8);
+INSERT INTO Bills
+VALUES (25, '2022-08-01',1,2);
+INSERT INTO Bills
+VALUES (642, '2022-06-01',0,3);
+INSERT INTO Bills
+VALUES (218, '2022-03-11',1,2);
+
+INSERT INTO ContractTypes
+VALUES('gas');
+
+INSERT INTO ContractTypes
+VALUES ('electricity');
+
+SELECT *
+FROM Contracts;
+SELECT *
+FROM Customers;
+SELECT *
+FROM Customers_Contracts;
+
+INSERT INTO Customers_Contracts
+VALUES(1,1);
+INSERT INTO Customers_Contracts
+VALUES(1,2);
+INSERT INTO Customers_Contracts
+VALUES(2,3);
+INSERT INTO Customers_Contracts
+VALUES(2,4);
+INSERT INTO Customers_Contracts
+VALUES(3,5);
+INSERT INTO Customers_Contracts
+VALUES(3,6);
+INSERT INTO Customers_Contracts
+VALUES(4,7);
+INSERT INTO Customers_Contracts
+VALUES(1,8);
+
+-- 3. Display all the customers' information ordered by the last name descending.
+
+SELECT *
+FROM Customers
+ORDER BY LastName DESC;
+
+-- 4. Display information about the bill with the highest amount.
+
+SELECT Id, Amount, Deadline, IsPaid, ContractId
+FROM Bills
+WHERE Amount =(SELECT MAX(Amount) FROM Bills);
+
+-- 5. Display the customers' information (just customer id, email address, street name, street number and city) together with their corresponding addresses.
+
+SELECT c.Id, c.Email, a.City, a.StreetName, a.StreetNumber
+FROM Customers c
+INNER JOIN Addresses a ON c.Id = a.CustomerId;
+
+-- 6. Pretend that one bill was paid today. Update it's corresponding information to reflect that.
+
+UPDATE Bills
+SET IsPaid = 1
+WHERE Id = 3;
+
+-- 7. Display the contract's information (contract id, start date, end date, bill id, amount, deadline, isPaid) together with their corresponding bills.
+
+SELECT c.Id 'Contract ID', c.StartDate, c.EndDate, b.Id 'Bill''s ID', b.Amount,b.Deadline, b.IsPaid
+FROM Contracts c
+INNER JOIN Bills b ON c.Id = b.ContractId;
+
+-- 8.Display the contract's information (contract id, start date, end date, contract type, bill id, amount, deadline, isPaid) together with their corresponding bills.
+
+SELECT c.Id 'Contract ID', c.StartDate, c.EndDate, ct.Name, b.Id 'Bill''s ID', b.Amount,b.Deadline, b.IsPaid
+FROM Contracts c
+    LEFT JOIN Bills b ON c.Id = b.ContractId
+    INNER JOIN ContractTypes ct ON ct.Id = c.Id;
+
+-- 9. Display the customers' information (just customer id, email address, contract end date & contract type) together with their corresponding contract type(s).
+
+SELECT cu.Id 'Customer ID', cu.Email, ct.EndDate, ct.ContractType 'ID Contract Type'
+FROM Customers_Contracts cc
+    INNER JOIN Customers cu ON cu.Id = cc.ContractId
+    INNER JOIN Contracts ct ON ct.Id = cc.CustomerId;
+
+-- 10. On exercise above make sure to display even the customers who don't have yet a contract.
+
+SELECT cu.Id 'Customer ID', cu.Email, ct.EndDate, ct.ContractType 'ID Contract Type'
```

Comment:
Your query returns the contracts that don't have a customer instead of returning the customers that don't have a contract.  Also, the comments above about the order of the table and about the mixing of customers and contracts on the join conditions also applies. Try this one: 
```
SELECT cu.Id 'Customer ID', cu.Email, ct.EndDate, ct.ContractType 'ID Contract Type'
FROM Customers cu
    LEFT JOIN Customers_Contracts cc ON cu.Id = cc.CustomerId
    LEFT JOIN Contracts ct ON ct.Id = cc.ContractId;
```